operator/builtin/input/k8sevent/k8s_event.go

```diff
@@ -81,25 +94,15 @@ func (k *K8sEvents) Start() error {
 		return errors.Wrap(err, "build client")
 	}
 
-	// Get all namespaces if left empty
-	if len(k.namespaces) == 0 {
-		k.namespaces, err = listNamespaces(ctx, k.client)
-		if err != nil {
-			return errors.Wrap(err, "list namespaces")
-		}
-	}
-
-	// Test connection
-	testWatcher, err := k.client.Events(k.namespaces[0]).Watch(ctx, metav1.ListOptions{})
```

Comment:
Can you think of any way other than testing a specific namespace? The reason I didn't like this is that it will only fail in some cases. If the operator is in discovery mode with no namespaces listed, it will do the exact behavior you described of silently logging and not causing the pod to fail. I'd rather have consistent behavior when starting up that's not dependent on the configuration.

Reply:
Maybe we wait to return until after we've run the first `listNamespaces` and after we've started all the watchers. A common issue is not giving the user `Watch` or `Events` permissions, so it seems like it would be valuable to be returning the errors from the actual calls.

Reply:
Will `listNamespaces` always return a populated array?